contracts/contracts/coordination/TACoChildApplication.sol

```diff
@@ -0,0 +1,122 @@
+    bytes32 public constant UPDATE_ROLE = keccak256("UPDATE_ROLE");
+    bytes32 public constant DEPLOYER_ROLE = keccak256("DEPLOYER_ROLE");
+
+    struct StakingProviderInfo {
+        address operator;
+        bool operatorConfirmed;
+        uint96 authorized;
+        // TODO: what about undelegations etc?
+    }
+
+    ITACoChildToRoot public immutable rootApplication;
+    address public coordinator;
+
+    mapping(address => StakingProviderInfo) public stakingProviderInfo;
+    mapping(address => address) public stakingProviderFromOperator;
+
+    constructor(ITACoChildToRoot _rootApplication, address[] memory updaters) {
+        require(
+            address(_rootApplication) != address(0),
+            "Address for root application must be specified"
+        );
+        rootApplication = _rootApplication;
+
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        for (uint256 i = 0; i < updaters.length; i++) {
+            _grantRole(UPDATE_ROLE, updaters[i]);
+        }
+    }
+
+    function setCoordinator(address _coordinator) external onlyRole(DEPLOYER_ROLE) {
+        require(coordinator == address(0), "Coordinator already set");
+        require(_coordinator != address(0), "Coordinator must be specified");
+        // require(_coordinator.numberOfRituals() >= 0, "Invalid coordinator");
+        coordinator = _coordinator;
+
+        // TODO reset role?
+    }
+
+    function authorizedStake(address _stakingProvider) external view returns (uint96) {
+        return stakingProviderInfo[_stakingProvider].authorized;
+    }
+
+    function updateOperator(
+        address stakingProvider,
+        address operator
+    ) external override onlyRole(UPDATE_ROLE) {
+        _updateOperator(stakingProvider, operator);
+    }
+
+    function updateAuthorization(
+        address stakingProvider,
+        uint96 amount
+    ) external override onlyRole(UPDATE_ROLE) {
+        _updateAuthorization(stakingProvider, amount);
+    }
+
+    function _updateOperator(address stakingProvider, address operator) internal {
+        StakingProviderInfo storage info = stakingProviderInfo[stakingProvider];
+        address oldOperator = info.operator;
+
+        if (operator != oldOperator) {
+            info.operator = operator;
+            // Update operator to provider mapping
+            stakingProviderFromOperator[oldOperator] = address(0);
+            stakingProviderFromOperator[operator] = stakingProvider;
+            info.operatorConfirmed = false;
+
+            emit OperatorUpdated(stakingProvider, operator);
+        }
+    }
+
+    function _updateAuthorization(address stakingProvider, uint96 amount) internal {
+        StakingProviderInfo storage info = stakingProviderInfo[stakingProvider];
+        uint96 fromAmount = info.authorized;
+
+        if (amount != fromAmount) {
+            info.authorized = amount;
+            emit AuthorizationUpdated(stakingProvider, amount);
```

Comment:
I noticed that we're emitting an event here. `AuthorizationUpdated`, but we're not emitting any events in `confirmOperatorAddress`. Whos is the consumer for the `AuthorizationUpdated` event?

Reply:
thanks I'll add event OperatorConfirmed

Reply:
apparently it was already added in that commit https://github.com/nucypher/nucypher-contracts/pull/114/commits/bef7fa1f7f1e4d3b8e11cda542a92fd6b73de416